code-challenges.js

```diff
@@ -41,35 +61,105 @@ var vowelTester1 = "learn"
 var vowelTester2 = "throw"
 // Expected output: 3
 
-
-
-
-
+//declare a function that will take in a string
+const firstVowel = (str) => {
+    //split string into an array
+    let stringArr = str.split("")
+    //create a variable that will take in vowels
+    //iterate through that string if the value is a vowel push that into new variable
+    let vowelsOnly = stringArr.filter(letter => letter.toLowerCase() === "a" || letter.toLowerCase() === "e" || letter.toLowerCase() === "i" || letter.toLowerCase() === "o" || letter.toLowerCase() === "u")
+    //store vowelsOnly into given string
+    let firstVowelInString = vowelsOnly[0]
+    //return the index of the first vowel in a word
+    let indexOfVowel = str.indexOf(firstVowelInString)
+    return indexOfVowel
+}
+    
+// console.log(firstVowel(vowelTester1))
+// console.log(firstVowel(vowelTester2))
 
 // --------------------5) Create a function that takes three arguments - two numbers and a mathematical operation (+, -, *, /) and performs the given calculation. If the input tries to divide by 0, return: "Can't divide by 0!"
 
-
-
-
-
+// establish a variable that stores all mathematical operations
+var operations = {
+    "+": function (num1, num2) {
+        return num1 + num2
+    },
+    "-": function (num1, num2) {
+        return num1 - num2
+    },
+    "*": function (num1, num2) {
+        return num1 * num2
+    },
+    "/": function (num1, num2) {
+        return num1 / num2
+    }
+}
+//create function that takes in two num arguments plus mathematical operations
+const calculator = (number1, operator, number2 ) => {
+    // If no numbers entered return error message
+    if(number1 === null || number2 === null) {
+        return "Enter yo digits first!"
+    } 
+    //if input is not a number return error messsage
+    if(isNaN(number1) || isNaN(number2)) {
+        return "I can only calculate numeros"
+    }
+    //if divided by zero return can't divide by 0!
+    if(number2 === 0 && operator === "/"){
+        return "Can't divide by 0!"
+    } 
+    //if can perform the given operation do so and return the result
+    return operations[operator](number1, number2) 
+}
 // Uncomment and use the following console logs to test your function
 // console.log(calculator(3, "*", 9))
-// Expected output: 27
-
+// // Expected output: 27
 // console.log(calculator(16, "+", 3))
-// Expected output: 19
-
+// // Expected output: 19
 // console.log(calculator(89, "/", 0))
-// Expected output: "Can't divide by 0!"
-
-
+// // Expected output: "Can't divide by 0!"
+// console.log(calculator("a", "-", "b"))
+// Expected output: "I can only calculate numeros"
```

Comment:
This solution is interesting! I'm curious as to why you chose to go this route?

Reply:
I was trying to figure out a way to turn the operators that were strings and turn them into something that actually does what the operator is. An example online had all the operators tied to their their own functions prior to the function to solve. I did run into some issues. It refused to move past my first "if statement" Didn't matter what I had in their first that would be the only thing that logged for all three cases ("Can't divide by 0!" x3, or 27 x3). I finally asked my brother for help who basically told me "Yeah, sometimes Javascript do be like that" and to make everything an if statement. He then went on to have me do the Null and NaN portions because he said it was good practice to make it as fullproof as possible. I had the logic he just gave me that last extra little tidbit.